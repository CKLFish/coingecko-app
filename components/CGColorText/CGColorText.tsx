import React from 'react';
import {Text, TextProps} from 'react-native';
import {useTheme} from 'react-native-paper';

import {TextUtil} from '../../Utils';

interface ColorTextProps extends Omit<TextProps, 'children'> {
  value: number;
  prefix?: string;
  postfix?: string;
}

const ColorText: React.FC<ColorTextProps> = props => {
  const {value, postfix = '%', prefix = '', style} = props;
  const {colors} = useTheme();

  return (
    <Text
      style={[
        {
          color:
            value > 0 ? colors.up : value === 0 ? colors.unchange : colors.down,
        },
        style,
      ]}>
      {`${prefix}${TextUtil.formatPercentage(value)}${postfix}`}
    </Text>
  );
};

export default ColorText;
