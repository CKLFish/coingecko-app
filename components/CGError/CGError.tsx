import React, {useCallback} from 'react';
import {useTranslation} from 'react-i18next';
import {StyleSheet, View} from 'react-native';
import {Button, Text} from 'react-native-paper';

type Props = {
  onPress?: () => void;
};

const CGError: React.FC<Props> = props => {
  const {onPress: propOnPress} = props;
  const {t} = useTranslation();

  const onPress = useCallback(() => {
    if (propOnPress) {
      propOnPress();
    }
  }, [propOnPress]);

  return (
    <View style={styles.container}>
      <View style={styles.inner}>
        <Text style={styles.centerText}>{t('NETWORK_FETCH_FAIL')}</Text>
        <Button onPress={onPress}>{t('NETWORK_FETCH_RETRY')}</Button>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexGrow: 1,
    flexDirection: 'row',
  },
  inner: {
    alignSelf: 'center',
    flex: 1,
  },
  centerText: {
    textAlign: 'center',
  },
});

export default CGError;
