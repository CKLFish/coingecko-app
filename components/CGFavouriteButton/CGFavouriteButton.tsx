import React, {useCallback} from 'react';
import {GestureResponderEvent} from 'react-native';
import {useTheme} from 'react-native-paper';
import {IconButtonProps} from 'react-native-vector-icons/Icon';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';

export interface FavouritebuttonProps
  extends Omit<IconButtonProps, 'color' | 'name'> {
  favourite?: boolean;
  onPressAddToFavourite?: (event: GestureResponderEvent) => void;
  onPressRemoveFavourite?: (event: GestureResponderEvent) => void;
}

const CGFavouriteButton: React.FC<FavouritebuttonProps> = props => {
  const {
    favourite,
    size,
    onPressRemoveFavourite: propOnPressRemoveFavourite,
    onPressAddToFavourite: propOnPressAddToFavourite,
    ...otherProps
  } = props;
  const {colors} = useTheme();

  const onPressAddToFavourite = useCallback(
    (event: GestureResponderEvent) => {
      if (propOnPressAddToFavourite) {
        propOnPressAddToFavourite(event);
      }
    },
    [propOnPressAddToFavourite],
  );

  const onPressRemoveFavourite = useCallback(
    (event: GestureResponderEvent) => {
      if (propOnPressRemoveFavourite) {
        propOnPressRemoveFavourite(event);
      }
    },
    [propOnPressRemoveFavourite],
  );

  return favourite ? (
    <MaterialCommunityIcons
      {...otherProps}
      name="heart-minus"
      size={size}
      color={colors.favourite}
      onPress={onPressRemoveFavourite}
    />
  ) : (
    <MaterialCommunityIcons
      {...otherProps}
      name="heart-plus-outline"
      size={size}
      onPress={onPressAddToFavourite}
    />
  );
};

export default CGFavouriteButton;
