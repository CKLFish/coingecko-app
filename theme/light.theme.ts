import {DefaultTheme} from 'react-native-paper';

const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    // # For Up / Down / Unchanged
    up: 'green',
    down: 'red',
    unchange: 'grey',
    favourite: 'red',
  },
};

export default theme;
