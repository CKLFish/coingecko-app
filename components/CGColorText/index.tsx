import CGColorText from './CGColorText';

export default CGColorText;
