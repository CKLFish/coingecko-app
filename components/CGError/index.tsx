import CGError from './CGError';

export default CGError;
