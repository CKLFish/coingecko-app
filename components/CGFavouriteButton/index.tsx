import CGFavouriteButton from './CGFavouriteButton';

export default CGFavouriteButton;
