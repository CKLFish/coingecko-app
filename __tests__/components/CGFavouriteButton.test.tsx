/**
 * @format
 */

import 'react-native';

import {shallow} from 'enzyme';
import React from 'react';
import {Provider} from 'react-native-paper';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import CGColorText from '../../components/CGColorText';
import CGError from '../../components/CGError';
import CGFavouriteButton from '../../components/CGFavouriteButton';
import theme from '../../theme/light.theme';

describe('CGFavouriteButton', () => {
  test('withCallback', () => {
    const favouriteButtonInstance = renderer.create(
      <CGFavouriteButton
        onPressAddToFavourite={() => {}}
        onPressRemoveFavourite={() => {}}
      />,
    );
    expect(favouriteButtonInstance.toJSON()).toMatchSnapshot();
  });
  test('favourite', () => {
    const favouriteButtonInstance = renderer.create(
      <CGFavouriteButton favourite={true} />,
    );
    expect(favouriteButtonInstance.toJSON()).toMatchSnapshot();
  });
  test('non-favourite', () => {
    const favouriteButtonInstance = renderer.create(
      <CGFavouriteButton favourite={false} />,
    );
    expect(favouriteButtonInstance.toJSON()).toMatchSnapshot();
  });
});
