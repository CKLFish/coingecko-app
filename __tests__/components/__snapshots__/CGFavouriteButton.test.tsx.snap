// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CGFavouriteButton favourite 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󱐯
</Text>
`;

exports[`CGFavouriteButton non-favourite 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󱐱
</Text>
`;

exports[`CGFavouriteButton withCallback 1`] = `
<Text
  allowFontScaling={false}
  onPress={[Function]}
  style={
    Array [
      Object {
        "color": undefined,
        "fontSize": 12,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  󱐱
</Text>
`;
